base/mpfr.jl

```diff
@@ -55,18 +56,29 @@ mutable struct BigFloat <: AbstractFloat
     sign::Cint
     exp::Clong
     d::Ptr{Limb}
+    # _d::Buffer{Limb} # Julia gc handle for memory @ d
+    _d::String # Julia gc handle for memory @ d (optimized)
```

Comment:
Mutating immutable memory is dangerous. If you must have this add a `FixedSizeBuffer` type.

Reply:
It's not _that_ dangerous, it's only bad if this pointer gets leaked anywhere else, but it's purely internal. I mean, it's not like there's a `deepcopy` method definition here that assumes they can't be mutated or anything like that. No, that would be crazy talk, and definitely not a bug in this very PR. :P

Reply:
Immutable memory is still being overwritten (since you construct it and then overwrite it with the result of an operation). I agree that it's much less likely to be harmful, since the compiler shouldn't be able to know the objectid of a newly-allocated bigfloat that gets subsequently modified (as long as the BigFloat itself isn't mutated more than once), but it does sound fairly risky.

Reply:
Isn't that just the same concern as any other usage of String?